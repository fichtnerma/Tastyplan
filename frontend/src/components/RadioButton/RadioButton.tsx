function RadioButton() {
    return <div></div>;
}

export default RadioButton;
