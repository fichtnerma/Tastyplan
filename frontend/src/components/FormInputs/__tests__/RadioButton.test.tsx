import React from 'react';
import { axe } from 'jest-axe';
import { render, fireEvent } from '@testing-library/react';
import RadioButton from '../Radiobutton';

describe('RadioButton Component', () => {
    const mockHandleChange = jest.fn();

    const radioBtn = {
        id: 'radioBtn1',
        label: 'Option 1',
        value: '1',
        checked: false,
    };

    const groupName = 'options';

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('renders without crashing', () => {
        const { container } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        expect(container).toMatchSnapshot();
    });

    it('renders radio button input', () => {
        const { getByRole } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        const radioInput = getByRole('radio');
        expect(radioInput).toBeInTheDocument();
        expect(radioInput).toHaveAttribute('name', groupName);
        expect(radioInput).toHaveAttribute('id', radioBtn.id);
    });

    it('renders label', () => {
        const { getByLabelText } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        const label = getByLabelText(radioBtn.label);
        expect(label).toBeInTheDocument();
    });

    it.skip('handles change', () => {
        const { getByRole } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        const radioInput = getByRole('radio');

        fireEvent.change(radioInput, { target: { checked: true } });

        expect(mockHandleChange).toHaveBeenCalledWith(radioBtn.id, true);
    });

    it('receives groupName prop', () => {
        const { getByRole } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        const radioInput = getByRole('radio');
        expect(radioInput).toHaveAttribute('name', groupName);
    });

    it.skip('receives radioBtn prop', () => {
        const { getByRole } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        const radioInput = getByRole('radio');
        expect(radioInput).toHaveAttribute('id', radioBtn.id);
        expect(radioInput).toHaveAttribute('defaultChecked', radioBtn.checked.toString());

        //const label = screen.getByLabelText(radioBtn.label);
        //expect(label).toBeInTheDocument();
    });

    it('receives handleChange prop', () => {
        const { getByRole } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        const radioInput = getByRole('radio');

        fireEvent.click(radioInput);

        expect(mockHandleChange).toHaveBeenCalledWith(radioBtn.id, true);
    });

    it('should not have basic accessibility issues', async () => {
        const { container } = render(
            <RadioButton groupName={groupName} radioBtn={radioBtn} handleChange={mockHandleChange} />,
        );
        const results = await axe(container);
        expect(results.violations).toHaveLength(0);
    });
});
