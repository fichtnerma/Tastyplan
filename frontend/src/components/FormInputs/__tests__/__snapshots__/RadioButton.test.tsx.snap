// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component renders without crashing 1`] = `
<div>
  <div
    class="checkboxContainer"
  >
    <input
      id="radioBtn1"
      name="options"
      role="radio"
      type="radio"
    />
    <label
      for="radioBtn1"
    >
      Option 1
    </label>
  </div>
</div>
`;
